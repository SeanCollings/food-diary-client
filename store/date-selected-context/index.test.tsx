import { FC, useEffect } from 'react';
import { act, render } from '@testing-library/react';
import {
  DateSelectedContextProvider,
  initialState,
  useDateSelectedContext,
} from '.';

const TestComponent = () => <div>Test component</div>;

const MockComponent: FC<{ update?: boolean }> = ({ update }) => {
  const { dateSelectedISO, updateSelectedDateISO } = useDateSelectedContext();

  useEffect(() => {
    if (update) {
      updateSelectedDateISO('01-01-2023');
    }
  }, [update, updateSelectedDateISO]);

  return <div>{dateSelectedISO}</div>;
};

describe('store - dates-selected-context', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should return initial state', () => {
    const { dateSelectedISO, updateSelectedDateISO } = initialState;

    expect(dateSelectedISO).toEqual('');
    expect(updateSelectedDateISO({} as any)).toBeNull();
  });

  it('should render provider with children', () => {
    const { asFragment } = render(
      <DateSelectedContextProvider>
        <TestComponent />
      </DateSelectedContextProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should return default date', async () => {
    const { asFragment } = render(
      <DateSelectedContextProvider>
        <MockComponent />
      </DateSelectedContextProvider>,
    );

    await act(() => Promise.resolve());

    expect(asFragment()).toMatchSnapshot();
  });

  it('should update state date', async () => {
    const { asFragment } = render(
      <DateSelectedContextProvider>
        <MockComponent update />
      </DateSelectedContextProvider>,
    );

    await act(() => Promise.resolve());

    expect(asFragment()).toMatchSnapshot();
  });
});
