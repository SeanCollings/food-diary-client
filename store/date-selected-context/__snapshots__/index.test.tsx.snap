// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store - dates-selected-context should render provider with children 1`] = `
<DocumentFragment>
  <div>
    Test component
  </div>
</DocumentFragment>
`;

exports[`store - dates-selected-context should return default date 1`] = `
<DocumentFragment>
  <div>
    2023-04-28
  </div>
</DocumentFragment>
`;

exports[`store - dates-selected-context should update state date 1`] = `
<DocumentFragment>
  <div>
    2023-01-01
  </div>
</DocumentFragment>
`;
